sympy/matrices/expressions/tests/test_matexpr.py

```diff
@@ -347,3 +349,11 @@ def test_factor_expand():
     assert expr1 != expr2
     assert expand(expr1) == expr2
     assert factor(expr2) == expr1
+
+
+def test_issue_7842():
```

Comment:
Maybe this could have a more informative name like `test_matrix_equality`

Reply:
I think that this is the recommended way of naming regression tests associated with an issue.

Reply:
I disagree with that in general. I think that putting issue numbers in comments as a paper trail is great but naming tests after issue numbers just means that all your tests have uninformative names.

Reply:
But if it ever fails then you can quickly find the related issue. The context (which the informative name would summarize) should be clear from the short test that was added.